keps/sig-storage/20191125-bucket-provisioning.md

```diff
@@ -222,25 +228,29 @@ Metadata:
   - cosi.io/finalizer [3]
 spec:
   bucketClassName: [4]
-  releasePolicy: {"Delete", "Retain"} [5]
-  bucketConfig: map[string]string [6]
-  bucketRef: [7]
+  supportedProtocols: [5]
+  releasePolicy: {"Delete", "Retain"} [6]
+  bucketConfig: map[string]string [7]
+  bucketRef: [8]
     name:
     namespace:
-  secretName: [8]
+  secretRef: [9]
+  accessMode: {"rw", "ro"} [10]
 status:
-  bucketMetaData: <map[string]string> [9]
-  phase: {"Bound", "Released", "Failed", "Errored"} [10]
+  bucketAttributes: <map[string]string> [11]
+  phase: {"Bound", "Released", "Failed", "Errored"} [12]
   conditions:
 ```
 1. `name`: Generated in the pattern of `“bucket-”<BUCKET-NAMESPACE>"-"<BUCKET-NAME>`
 1. `labels`: central controller adds the label to its managed resources for easy CLI GET ops.  Value is the driver name returned by GetDriverInfo() rpc*.
 1. `finalizers`: COSI controller adds the finalizer to defer Bucket deletion until backend deletion ops succeed.
 1. `bucketClassName`: Name of the associated BucketClass
+1. `supportedProtocols`:  String array of protocols (e.g. s3, gcs, swift, etc.) supported by the associated object store.
```

Comment:
I agree that the BC's supportedProtocols should be a list, and maybe the idea is that list is copied here. In that case maybe under `status` we need something like `actualProtocol`?

Reply:
For greenfield it's mostly a bookkeeping field.
For brownfield/static I think it'd be used to match Buckets.  As such, an admin would define this field, meaning it'd be in spec.

Reply:
Do we keep track of the protocol actually used? If I have a list of acceptable protocols do I care which one was bound to? Not sure but maybe?

Reply:
> Do we keep track of the protocol actually used?

As best we can.  I think the GetDriverInfo will contain this information, in addition to the driver name.